pkg/controller/defaults/placements.go

```diff
@@ -35,6 +35,28 @@ var (
 			},
 		},
 
+		"mon": rook.Placement{
+			PodAntiAffinity: &corev1.PodAntiAffinity{
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+					corev1.WeightedPodAffinityTerm{
+						Weight: 100,
+						PodAffinityTerm: corev1.PodAffinityTerm{
+							LabelSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
+									metav1.LabelSelectorRequirement{
+										Key:      "app",
+										Operator: metav1.LabelSelectorOpIn,
+										Values:   []string{"rook-ceph-mon"},
+									},
+								},
+							},
+							TopologyKey: "kubernetes.io/hostname",
+						},
+					},
+				},
+			},
+		},
```

Comment:
I have some reservations about this approach (of  consolidating PodAntiAffinity into `all`). 

1. Since label selector is not same for all the components (`mons`, `mds` etc), this consolidation would mean using a place holder for the label selector  in the podAntiAffinity  (or entire placement config)for `all` (which would indeed be overridden for different components).  And looking at the rook (https://rook.io/docs/rook/v1.2/ceph-cluster-crd.html#placement-configuration-settings)  documents, `all` behaves like a generic placement configuration that gets applied to other components (unless they are overridden), so this would mean that the placeholder in `all` would have be overridden by the all components. 

2. Looking at the existing placements configurations (like mds, rgw, etc), they also use the same placement configuration but still displayed separately (and not consolidated with `all`).

@jarrpa @umangachapagain

Reply:
While I still think it would be a reasonable thing to do, I'll say it is out of scope for this PR to do that consolidation at this time. we'll pick it up later.

Reply:
The mon anti-affinfity makes sense under `mon`. If we move it to `all`, the other daemons would also get anti-affinity with mons. Why would we want that?